File: 0121 - Best Time To Buy and Sell Stock/BestTimeToBuyAndSellStock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
  let min = Number.MAX_VALUE;
  let profit = 0;

  prices.forEach((price) => {
    min = Math.min(min, price);
    profit = Math.max(profit, price - min);
  });

  return profit;
};

File: 0121 - Best Time To Buy and Sell Stock/BestTimeToBuyAndSellStock.go
import (
    "math"
)

func maxProfit(prices []int) int {
    min := math.MaxInt32
    profit := 0
    
    for i := 0; i < len(prices); i++ {
        if prices[i] < min {
            min = prices[i]
        } else if profit < prices[i] - min {
            profit = prices[i] - min
        }
    }
    
    return profit
}